Polyfication with variable dynamic import not working

Hello

First of all, thank you for the library

## Which package?

This is true for every package that handle polyfication and expose some `locale-data`:

- `@formatjs/intl-datetimeformat"` (the one I use in my repro)
- `@formatjs/intl-displaynames"`
- `@formatjs/intl-listformat`
- `@formatjs/intl-numberformat`
- `@formatjs/intl-pluralrules`
- `@formatjs/intl-relativetimeformat`

## Describe the bug

When having a variable dynamic import causes bundling errors.

```ts
await import(`@formatjs/intl-datetimeformat/locale-data/${locale}.js`)
```

I will cause this kind of error : 

```txt
Failed to compile.

Module not found: Error: "./locale-data" is not exported under the conditions ["import","module","webpack","production","browser"] from package /Users/romain.trotard/mirakl/workspace/repro/formatjs-dynamic-import/node_modules/@formatjs/intl-datetimeformat (see exports field in /Users/romain.trotard/mirakl/workspace/repro/formatjs-dynamic-import/node_modules/@formatjs/intl-datetimeformat/package.json)
```

## To Reproduce

### Codesandbox URL

I have done a repository that you can clone https://github.com/romain-trotard/formatjs-dynamic-import

### Reproducible Steps/Repo

Steps to reproduce the behavior:

1. `npm i`
2. `npm run build`

## Expected behavior

Be able to build when template literal dynamic import.

## Explanation of the problem

We can see that from version `7.0.0` of `@formatjs/intl-datetimeformat` is exposing thanks to `exports` property:

```json
{
  "exports": {
    ".": "./index.js",
    "./polyfill.js": "./polyfill.js",
    "./polyfill-force.js": "./polyfill-force.js",
    "./should-polyfill.js": "./should-polyfill.js",
    "./add-all-tz.js": "./add-all-tz.js",
    "./add-golden-tz.js": "./add-golden-tz.js",
    "./locale-data/*": "./locale-data/*"
  },
}
```

> **Note:** The commit is the following one https://github.com/formatjs/formatjs/commit/c717c1adee58ad9d7604c74a4bafca2628e2a0a8

The problem comes when projects want to dynamically import a variable locale:

```ts
await import(`@formatjs/intl-datetimeformat/locale-data/${locale}.js`)
```

> **Note:** You can see the code in the `App.tsx` file.

In this case, `webpack` will try to see if `@formatjs/intl-datetimeformat/locale-data` is accessible because `webpack` cannot know what will be the `locale` value -> the first non dynamic part.

But the problem is that each package, does not expose the `locale-data` folder only.

> **Note:** `"./locale-data/*": "./locale-data/*"` is nice if the dynamic import has no variable.
> 
> i.e. `await import('@formatjs/intl-datetimeformat/locale-data/en.js');` works fine.

## Solution

The solution is to expose the `locale-data` folder alone too.


```json
{
  "exports": {
    ".": "./index.js",
    "./polyfill.js": "./polyfill.js",
    "./polyfill-force.js": "./polyfill-force.js",
    "./should-polyfill.js": "./should-polyfill.js",
    "./add-all-tz.js": "./add-all-tz.js",
    "./add-golden-tz.js": "./add-golden-tz.js",
    "./locale-data/*": "./locale-data/*",
    "./locale-data": "./locale-data"
  },
}
```


## Additional information

A similar issue has been opened in 2023 when this `exports` has been tried but caused the same issue https://github.com/formatjs/formatjs/issues/4127 and changes seem to have been reverted

For the moment I do not upgrade my libraries, and stay on the versions without `exports`.


## Desktop:

- OS: macOS

--- a/packages/intl-datetimeformat/package.json
+++ b/packages/intl-datetimeformat/package.json
@@ -13,7 +13,8 @@
     "./should-polyfill.js": "./should-polyfill.js",
     "./add-all-tz.js": "./add-all-tz.js",
     "./add-golden-tz.js": "./add-golden-tz.js",
-    "./locale-data/*": "./locale-data/*"
+    "./locale-data/*": "./locale-data/*",
+    "./locale-data": "./locale-data"
   },
   "dependencies": {
     "@formatjs/ecma402-abstract": "workspace:*",

--- a/packages/intl-displaynames/package.json
+++ b/packages/intl-displaynames/package.json
@@ -11,7 +11,8 @@
     "./polyfill.js": "./polyfill.js",
     "./polyfill-force.js": "./polyfill-force.js",
     "./should-polyfill.js": "./should-polyfill.js",
-    "./locale-data/*": "./locale-data/*"
+    "./locale-data/*": "./locale-data/*",
+    "./locale-data": "./locale-data"
   },
   "dependencies": {
     "@formatjs/ecma402-abstract": "workspace:*",

--- a/packages/intl-listformat/package.json
+++ b/packages/intl-listformat/package.json
@@ -11,7 +11,8 @@
     "./polyfill.js": "./polyfill.js",
     "./polyfill-force.js": "./polyfill-force.js",
     "./should-polyfill.js": "./should-polyfill.js",
-    "./locale-data/*": "./locale-data/*"
+    "./locale-data/*": "./locale-data/*",
+    "./locale-data": "./locale-data"
   },
   "dependencies": {
     "@formatjs/ecma402-abstract": "workspace:*",

--- a/packages/intl-numberformat/package.json
+++ b/packages/intl-numberformat/package.json
@@ -10,7 +10,8 @@
     "./polyfill.js": "./polyfill.js",
     "./polyfill-force.js": "./polyfill-force.js",
     "./should-polyfill.js": "./should-polyfill.js",
-    "./locale-data/*": "./locale-data/*"
+    "./locale-data/*": "./locale-data/*",
+    "./locale-data": "./locale-data"
   },
   "dependencies": {
     "@formatjs/ecma402-abstract": "workspace:*",

--- a/packages/intl-pluralrules/package.json
+++ b/packages/intl-pluralrules/package.json
@@ -11,7 +11,8 @@
     "./polyfill.js": "./polyfill.js",
     "./polyfill-force.js": "./polyfill-force.js",
     "./should-polyfill.js": "./should-polyfill.js",
-    "./locale-data/*": "./locale-data/*"
+    "./locale-data/*": "./locale-data/*",
+    "./locale-data": "./locale-data"
   },
   "dependencies": {
     "@formatjs/ecma402-abstract": "workspace:*",

--- a/packages/intl-relativetimeformat/package.json
+++ b/packages/intl-relativetimeformat/package.json
@@ -11,7 +11,8 @@
     "./polyfill.js": "./polyfill.js",
     "./polyfill-force.js": "./polyfill-force.js",
     "./should-polyfill.js": "./should-polyfill.js",
-    "./locale-data/*": "./locale-data/*"
+    "./locale-data/*": "./locale-data/*",
+    "./locale-data": "./locale-data"
   },
   "dependencies": {
     "@formatjs/ecma402-abstract": "workspace:*",

